source/data-formats/extended-json.txt

```diff
@@ -0,0 +1,219 @@
+
+         {
+           "_id": { "$oid": "573a1391f29313caabcd9637" },
+           "createdAt": { "$date": "2020-09-30T18:22:51.648Z" },
+           "numViews": 36520312
+         }
+
+Write Extended JSON
+-------------------
+
+You can write an Extended JSON string from a BSON document object by using the
+``EJSON.stringify()`` method.
+
+The following example outputs an Extended JSON string in the Relaxed format:
+
+.. io-code-block::
+
+   .. input::
+      :language: js
+
+      import { Binary, Code, BSON } from 'mongodb';
+      const EJSON = BSON.EJSON;
+
+      const doc = {
+         foo: [1, 2],
+         bar: { hello: "world" },
+         code: new Code("function x() { return 1; }", {}),
+         date: new Date(2024, 6, 20, 10, 30, 0),
+      };
+
+      const ejsonStr = EJSON.stringify(doc);
+      console.log(ejsonStr);
+
+   .. output::
+      :language: none
+      :visible: false
+
+      [{"foo":[1,2]},{"bar":{"hello":"world"}},{"code":{"$code":"function x() { return 1; }","$scope":{}}},{"date":{"$date":"2024-07-20T14:30:00Z"}}]
+
+By default, the ``stringify()`` method returns the Extended JSON string in the 
+Relaxed format. To specify the Canonical format, set the ``relaxed`` 
+option to ``false``.
+
+The following example shows how to output Extended JSON in the Canonical format:
+
+.. io-code-block::
+
+   .. input::
+      :language: js
+
+      const { Binary, Code, EJSON } = require('bson');
+
+      const doc = {
+         foo: [1, 2],
+         bar: { hello: "world" },
+         code: new Code("function x() { return 1; }", {}),
+         date: new Date(2024, 6, 20, 10, 30, 0),
+      };
+
+      const ejsonStr = EJSON.stringify(doc, { relaxed: false });
+      print(ejsonStr)
+
+   .. output::
+      :language: none
+      :visible: false
+
+      [{"foo":[{"$numberInt":"1"},{"$numberInt":"2"}]},{"bar":{"hello":"world"}},{"code":{"$code":"function x() { return 1; }","$scope":{}}},{"date":{"$date":{"$numberLong":"1721485800000"}}}]
+
+Read Extended JSON
+------------------
+
+You can read an Extended JSON string into the JavaScript value or object described 
+by the string by using the ``EJSON.parse()`` method. 
+
+The following example shows how you can read an Extended JSON string into a
+JavaScript value or object by using the ``parse()`` method:
+
+.. The back ticks on ejsonStr in the below code example causes the orange 
+.. highlighting in this file, but the page still renders correctly.
+
+.. io-code-block::
+
+   .. input::
+      :language: javascript
+
+      import { BSON } from 'mongodb';
+      const EJSON = BSON.EJSON;
+
+      const ejsonStr = `{
+         "foo": [
+            { "$numberInt": "1" },
+            { "$numberInt": "2" }
+         ],
+         "bar": { "hello": "world" },
+         "code": {
+            "$code": "function x() { return 1; }",
+            "$scope": {}
+         },
+         "bin": { "$binary": { "base64": "AQIDBA==", "subType": "00" } }
+         }`;
```

Comment:
```suggestion
       }`;
```

Doesn't functionally make a difference but is the proper indentation level for this closing brace